src/store/user-store.js

```diff
@@ -1,16 +1,48 @@
-import { makeAutoObservable } from "mobx";
+import { makeAutoObservable, flow } from "mobx";
```

Comment:
Step 1: we need to add `flow` to the store by importing it.

Read more about `flow` [here](https://mobx.js.org/actions.html#using-flow-instead-of-async--await-)